### PR TITLE
[WIP] Swagger Router Documentation

### PR DESCRIPTION
**Pull request by Joannis**

### CODE REVIEW

File: Sources/Swagger/OpenAPI.swift [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        return String(bytes: try JSONEncoder().encode(self), encoding: .utf8)
+    }
+    
+    public func write(to file: URL) throws {
+        let data = try JSONEncoder().encode(self)
+        try data.write(to: file)
+    }
+}
+
+public enum Error: Swift.Error {
```
Comment (tanner0101) [inline Sources/Swagger/OpenAPI.swift]: we should avoid public enum errors wherever possible

File: Tests/SwaggerTests/SwaggerTests.swift [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        
+        router.on(.get, to: "hello", "world") { _ in
+            return try Response(body: "Hello world")
+        }.describe(as: """
+            Prints "Hello world" using a `200` OK status code
+        """) { _ in
+            return Swagger.Operation(response: "Hello world")
+        }
+        
+        try router.describeAPI(named: "example").write(to: URL(fileURLWithPath: "/Users/joannisorlandos/test.json"))
```
Comment (tanner0101) [inline Tests/SwaggerTests/SwaggerTests.swift]: naming routes could be really useful for generating links to the routes as well (something laravel does)

**Comment by bre7** [comment]
Was this feature abandoned ? Something like Kitura's plugin would be pretty cool: https://github.com/IBM-Swift/Kitura-OpenAPI

**Comment by tyoc213** [comment]
Something like https://github.com/tiangolo/fastapi#interactive-api-docs

**Comment by joscdk** [comment]
It's planned to make a flexible integration in the Router for Vapor 4, that will make it possible to make integrations like these easy.

It's planned the `vapor/open-api` will be build against Vapor 4 after Vapor 4 is released :)

**Comment by tyoc213** [comment]
It is to much to ask about the ETA to version 4?

**Comment by joscdk** [comment]
@tyoc213 there aren't currently an ETA, however i can say so much, not before Swift 5.1, since Vapor 4 will require Swift 5.1 :)

**Pull request closed without merging**